### PR TITLE
eos-repo-server: Reset timeout conditions on failed requests

### PR DESCRIPTION
**Pull request by pwithnall**
Previously they were reset only on successful requests, meaning that a
long-running timed out request in the middle of a legitimate stream of
update requests from a client would cause eos-update-server to exit.

Signed-off-by: Philip Withnall <withnall@endlessm.com>

https://phabricator.endlessm.com/T16270

### CODE REVIEW

File: src/eos-repo-server.c [hunk 1]
```diff
@@ -830,9 +831,6 @@ server_cb (SoupServer *soup_server,
   g_object_get (msg,
                 "status-code", &status_code,
                 NULL);
```
Comment (cosimoc) [inline src/eos-repo-server.c]: Does not look like `status_code` is used anymore.

File: src/eos-repo-server.c [hunk 2]
```diff
@@ -403,20 +403,29 @@ eos_filez_read_data_disconnect_and_clear_msg (EosFilezReadData *read_data)
...
-                         guint new_value)
+                         gint                  delta)
 {
-  if (server->pending_requests == new_value)
+  GObject *obj = G_OBJECT (server);
+
+  g_assert (delta >= 0 || server->pending_requests >= (guint) -delta);
+  g_assert (delta <= 0 || server->pending_requests <= G_MAXUINT - delta);
+
+  if (delta == 0)
```
Comment (cosimoc) [inline src/eos-repo-server.c]: Any reason not to update `server->last_request_time` in this case?

File: src/eos-repo-server.c (hunk 1 above)
Comment (pwithnall) [inline src/eos-repo-server.c]: Whoops.

File: src/eos-repo-server.c (hunk 2 above)
Comment (pwithnall) [inline src/eos-repo-server.c]: Because the number of requests hasn’t changed, so there is no new ‘last request’ to change the time for.

**Comment by pwithnall** [comment]
Updated branch to address review comments.

**Pull request merged**